maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java

```diff
@@ -261,8 +261,7 @@ else if ( request.isUpdateSnapshots() )
         }
         session.setAuthenticationSelector( authSelector );
 
-        String transport = request.getUserProperties()
-                .getProperty( MAVEN_RESOLVER_TRANSPORT_KEY, MAVEN_RESOLVER_TRANSPORT_WAGON );
```

Comment:
So this is a merged, final version of the properties and the actual source is irrelevant?

Reply:
yup, this is "config" properties (merged plus added things). I think I understand from where "request.systemProperties" are sourced, but from where are "request.userProperties" sourced?

Reply:
Look into Maven CLI, that is the source for it. Commons CLI provides it and it is passed down the line.

Reply:
`request.*Properties` is populated in 
https://github.com/apache/maven/blob/master/maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java